src/pykeen/triples/triples_factory.py

```diff
@@ -412,7 +412,7 @@ def __init__(
             num_relations=num_relations,
             create_inverse_triples=create_inverse_triples,
         )
-        self.mapped_triples = mapped_triples
+        self.mapped_triples = torch.as_tensor(mapped_triples, dtype=torch.long)
```

Comment:
should we do some integrity/type checks here? like make sure that if it's a np.ndarray it should have certain dtype?

Reply:
also, what's the consequence of not having the tensor on a specific device?

Reply:
> also, what's the consequence of not having the tensor on a specific device?

it will be on cpu (except maybe it would already be a tensor, and on a different device)